frontend/src/components/SpotCard/SpotCard.jsx

```diff
@@ -1,31 +1,43 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { FaStar } from 'react-icons/fa';
 import './SpotCard.css';
 
 function SpotCard({ spot }) {
     const [showTooltip, setShowTooltip] = useState(false);
+    const [timer, setTimer] = useState(null)
 
 
     const handleMouseEnter = () => {
-        setShowTooltip(true);
+        const newTimer = setTimeout(() => {
+            setShowTooltip(true);
+        }, 500);
+        setTimer(newTimer);
     };
 
     const handleMouseLeave = () => {
+        clearTimeout(timer);
         setShowTooltip(false);
     };
 
+    useEffect(() => {
+        return () => {
+            clearTimeout(timer);
+        }
+    }, [timer]);
+
+    let avgRating = "New!";
+
+    if (spot.avgRating)
+        avgRating = spot.avgRating.toString().slice(0, 3);
```

Comment:
In the future, data manipulation like this (including the "new" functionality) should all happen on the backend